src/parquet/file_writer.h

```diff
@@ -100,7 +118,7 @@ class PARQUET_EXPORT ParquetFileWriter {
     /// \note Deprecated since 1.3.0
     RowGroupWriter* AppendRowGroup(int64_t num_rows);
 
-    virtual RowGroupWriter* AppendRowGroup() = 0;
+    virtual RowGroupWriter* AppendRowGroup(bool row_group_by_size = false) = 0;
```

Comment:
Adding a new method `AppendFixedSizeRowGroup()` would result in more readable code than `AppendRowGroup(true)`